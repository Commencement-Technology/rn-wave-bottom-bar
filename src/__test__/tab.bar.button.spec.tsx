import React from 'react';
import { View } from 'react-native';
import * as renderer from 'react-test-renderer';
import {
  withReanimatedTimer,
  advanceAnimationByTime,
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
} from 'react-native-reanimated/lib/reanimated2/jestUtils';

import { BarButton, TabBarButton } from '../components/tab.bar.button';

const tabBarIcon = (props: any) => <View {...props} />;

describe('TabBarButton', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });
  afterAll(() => {
    jest.useRealTimers();
  });
  it("Should render well when it's Focus", () => {
    withReanimatedTimer(() => {
      const tree = renderer.create(
        <BarButton
          mode="default"
          index={0}
          isFocused
          options={{ tabBarIcon }}
          onPress={jest.fn()}
          onLongPress={jest.fn()}
        />
      );
      advanceAnimationByTime(500);
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
  it('Should render well it is not focused', () => {
    withReanimatedTimer(() => {
      const tree = renderer.create(
        <BarButton
          mode="default"
          index={0}
          isFocused={false}
          options={{ tabBarIcon }}
          onPress={jest.fn()}
          onLongPress={jest.fn()}
        />
      );
      advanceAnimationByTime(500);
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
  it("Should render well when it's Focus", () => {
    withReanimatedTimer(() => {
      const tree = renderer.create(
        <TabBarButton
          mode="default"
          index={0}
          isFocused
          options={{ tabBarIcon }}
          onPress={jest.fn()}
          onLongPress={jest.fn()}
        />
      );
      advanceAnimationByTime(500);
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
  it('Should render well it is not focused', () => {
    withReanimatedTimer(() => {
      const tree = renderer.create(
        <TabBarButton
          mode="default"
          index={0}
          isFocused={false}
          options={{ tabBarIcon }}
          onPress={jest.fn()}
          onLongPress={jest.fn()}
        />
      );
      advanceAnimationByTime(500);
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
  it('Should render well it is not focused', () => {
    withReanimatedTimer(() => {
      const tree = renderer.create(
        <TabBarButton
          mode="square"
          index={0}
          isFocused={false}
          options={{ tabBarIcon }}
          onPress={jest.fn()}
          onLongPress={jest.fn()}
        />
      );
      advanceAnimationByTime(500);
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
