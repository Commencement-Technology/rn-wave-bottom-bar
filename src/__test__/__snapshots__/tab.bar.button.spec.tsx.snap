// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarButton Should render well it is not focused 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "elevation": 12,
      "flex": 1,
      "justifyContent": "center",
      "zIndex": 1,
    }
  }
>
  <View
    animatedStyle={
      {
        "value": {
          "opacity": 0,
          "transform": [
            {
              "scale": 0,
            },
          ],
        },
      }
    }
    collapsable={false}
    style={
      {
        "opacity": 0,
        "transform": [
          {
            "scale": 0,
          },
        ],
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "elevation": 12,
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
          "zIndex": 12,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "zIndex": 12,
          }
        }
      >
        <View
          color="white"
          focused={false}
          size={28}
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`TabBarButton Should render well it is not focused 2`] = `
<View
  style={
    {
      "alignItems": "center",
      "elevation": 12,
      "flex": 1,
      "justifyContent": "center",
      "zIndex": 1,
    }
  }
>
  <View
    animatedStyle={
      {
        "value": {
          "transform": [
            {
              "translateY": -18,
            },
          ],
        },
      }
    }
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "green",
        "borderRadius": 32,
        "elevation": 10,
        "height": 60,
        "justifyContent": "center",
        "position": "absolute",
        "transform": [
          {
            "translateY": -18,
          },
        ],
        "width": 60,
        "zIndex": -1,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 32,
          "elevation": 10,
          "height": 60,
          "justifyContent": "center",
          "opacity": 1,
          "position": "absolute",
          "width": 60,
          "zIndex": -1,
        }
      }
    >
      <View
        color="white"
        focused={false}
        size={28}
      />
    </View>
  </View>
</View>
`;

exports[`TabBarButton Should render well it is not focused 3`] = `
<View
  style={
    {
      "alignItems": "center",
      "elevation": 12,
      "flex": 1,
      "justifyContent": "center",
      "zIndex": 1,
    }
  }
>
  <View
    animatedStyle={
      {
        "value": {
          "transform": [
            {
              "translateY": -18,
            },
          ],
        },
      }
    }
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "green",
        "borderRadius": 32,
        "elevation": 10,
        "height": 60,
        "justifyContent": "center",
        "position": "absolute",
        "transform": [
          {
            "translateY": -18,
          },
        ],
        "width": 60,
        "zIndex": -1,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 16,
          "elevation": 10,
          "height": 60,
          "justifyContent": "center",
          "opacity": 1,
          "position": "absolute",
          "width": 60,
          "zIndex": -1,
        }
      }
    >
      <View
        color="white"
        focused={false}
        size={28}
      />
    </View>
  </View>
</View>
`;

exports[`TabBarButton Should render well when it's Focus 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "elevation": 12,
      "flex": 1,
      "justifyContent": "center",
      "zIndex": 1,
    }
  }
>
  <View
    animatedStyle={
      {
        "value": {
          "opacity": 0,
          "transform": [
            {
              "scale": 0,
            },
          ],
        },
      }
    }
    collapsable={false}
    style={
      {
        "opacity": 0,
        "transform": [
          {
            "scale": 0,
          },
        ],
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "elevation": 12,
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
          "zIndex": 12,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "zIndex": 12,
          }
        }
      >
        <View />
      </View>
    </View>
  </View>
</View>
`;

exports[`TabBarButton Should render well when it's Focus 2`] = `
<View
  style={
    {
      "alignItems": "center",
      "elevation": 12,
      "flex": 1,
      "justifyContent": "center",
      "zIndex": 1,
    }
  }
>
  <View
    animatedStyle={
      {
        "value": {
          "transform": [
            {
              "translateY": -18,
            },
          ],
        },
      }
    }
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "green",
        "borderRadius": 32,
        "elevation": 10,
        "height": 60,
        "justifyContent": "center",
        "position": "absolute",
        "transform": [
          {
            "translateY": -18,
          },
        ],
        "width": 60,
        "zIndex": -1,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 32,
          "elevation": 10,
          "height": 60,
          "justifyContent": "center",
          "opacity": 1,
          "position": "absolute",
          "width": 60,
          "zIndex": -1,
        }
      }
    >
      <View
        color="white"
        focused={true}
        size={28}
      />
    </View>
  </View>
</View>
`;
